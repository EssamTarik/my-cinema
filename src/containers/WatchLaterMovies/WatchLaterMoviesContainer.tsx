import React from "react";
import { IProps } from "./interfaces";
import MovieGridContainer from "../MovieGrid";
import { IMovie } from "../../store/movieSearch/models";

const FavoriteMoviesContainer = ({ watchLater }: IProps) => {
  const favoriteMovies = Object.values(watchLater) as IMovie[];
  return <MovieGridContainer movies={favoriteMovies} />;
};

export default FavoriteMoviesContainer;
