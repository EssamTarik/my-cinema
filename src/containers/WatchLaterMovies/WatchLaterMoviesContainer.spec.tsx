jest.mock('../MovieGrid', () => () => <div />)
import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
const mockMovie = jest.requireMock('movie');

import WatchLaterMoviesContainer from "./WatchLaterMoviesContainer";

describe('WatchLaterMoviesContainer', () => {
  afterEach(cleanup);
  it("renders", () => {
    const { asFragment } = render(
      <WatchLaterMoviesContainer watchLater={{ [mockMovie.id]: mockMovie }} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
})

