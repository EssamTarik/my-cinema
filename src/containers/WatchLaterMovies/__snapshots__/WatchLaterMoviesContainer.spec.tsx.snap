// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatchLaterMoviesContainer renders 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
